### PR TITLE
add mobx-react w/ npm auto-update

### PR DESCRIPTION
**Pull request by kennynaoh**
Pull request for issue: #8309 
@pvnr0082t , @x09326 please help me review this pull request, thanks!
Manually add other versions from git repo because some of old versions can't get from npm. Update .gitignore to ignore some versions with unusual version name.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.
Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/mobxjs/mobx-react
 * Official website (optional, not the repository):
 * NPM package url (optional): https://www.npmjs.com/package/mobx-react
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 419
   - Count of watchers: 25
   - Count of forks: 42
 * NPM download stats (optional):
   - Downloads in the last day: 1782
   - Downloads in the last week: 10485
   - Downloads in the last month: 37619

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by pvnr0082t** [comment]
LGTM

File: ajax/libs/mobx-react/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "react",
+    "reactjs",
+    "reactive"
+  ],
+  "npmName": "mobx-react",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "+(index|custom|native).+(js|map)"
```
Comment (x09326) [inline ajax/libs/mobx-react/package.json]: @kennynaoh 
Why do you need map here?
I saw there is no map file in the npm package.

Comment (PeterDaveHello) [inline ajax/libs/mobx-react/package.json]: for feature compatible

**Comment by x09326** [comment]
LGTM
ping @PeterDaveHello

**Comment by PeterDaveHello** [comment]
Thanks!

**Pull request merged**